packages/monitors/src/kpi-monitor.ts

```diff
@@ -0,0 +1,330 @@
+      );
+      const usdValue = usdRate
+        ? localValue
+            .mul(usdRate.rate)
+            .div(BigNumber.from(10).pow(usdRate.decimals))
+        : localValue;
+      switch (v.tradeType.toLowerCase()) {
+        case 'lend':
+          acc.lend = acc.lend.add(usdValue);
+          break;
+        case 'borrow':
+          acc.borrow = acc.borrow.add(usdValue);
+          break;
+
+        default:
+          break;
+      }
+      acc.total = acc.total.add(usdValue);
+      acc.decimals = usdRate?.decimals ?? 8;
+
+      return acc;
+    },
+    {
+      lend: BigNumber.from(0),
+      borrow: BigNumber.from(0),
+      total: BigNumber.from(0),
+    }
+  );
+}
+
+async function processAccountResults(accounts: Account[]) {
+  try {
+    const nowSeconds = new Date().getTime() / 1000;
+    const dailyActive = accounts.filter(
+      (a) => a.lastUpdateTimestamp > nowSeconds - secondsInDay
+    );
+    const weeklyActive = accounts.filter(
+      (a) => a.lastUpdateTimestamp > nowSeconds - secondsInDay * 7
+    );
+    const monthlyActive = accounts.filter(
+      (a) => a.lastUpdateTimestamp > nowSeconds - secondsInDay * 30
+    );
+
+    kpis.accounts = {
+      total: {
+        overall: accounts.length,
+        borrowers: accounts.filter(
+          (a) => a.hasCashDebt || a.hasPortfolioAssetDebt
+        ).length,
+      },
+      daily: {
+        overall: dailyActive.length,
+        borrowers: dailyActive.filter(
+          (a) => a.hasCashDebt || a.hasPortfolioAssetDebt
+        ).length,
+      },
+      weekly: {
+        overall: weeklyActive.length,
+        borrowers: weeklyActive.filter(
+          (a) => a.hasCashDebt || a.hasPortfolioAssetDebt
+        ).length,
+      },
+      monthly: {
+        overall: monthlyActive.length,
+        borrowers: monthlyActive.filter(
+          (a) => a.hasCashDebt || a.hasPortfolioAssetDebt
+        ).length,
+      },
+    };
+
+    Object.keys(kpis.accounts).forEach((period) => {
+      const { overall, borrowers } = kpis.accounts[period];
+      const overallSeries = {
+        metric: `kpi.accounts.${period}.overall}`,
+        tags: [`period:${period}`, `type:overall`, `chain:mainnet`],
+        points: [
+          {
+            timestamp: Math.round(new Date().getTime() / 1000),
+            value: overall,
+          },
+        ],
+      };
+      const borrowerSeries = {
+        metric: `kpi.accounts.${period}.borrowers`,
+        tags: [`period:${period}`, `type:borrowers`, `chain:mainnet`],
+        points: [
+          {
+            timestamp: Math.round(new Date().getTime() / 1000),
+            value: borrowers,
+          },
+        ],
+      };
+      series.push(overallSeries, borrowerSeries);
+    });
+  } catch (e) {
+    console.log(e);
+    await log({
+      message: (e as Error).message,
+      level: 'error',
+      chain: 'mainnet',
+      action: 'kpi.accounts',
+    });
+  }
+}
+
+function setExchangeRatesUSD(rates: ExchangeRate[]) {
```

Comment:
This looks ok to me but I think that a unit test or something would make it more clear that the outputs are correct.

Reply:
I'll add unit tests as part of cleanup.